pkgs/applications/networking/instant-messengers/psi-plus/default.nix

```diff
@@ -21,20 +43,30 @@ mkDerivation rec {
   };
 
   cmakeFlags = [
-    "-DENABLE_PLUGINS=ON"
-  ];
-
-  nativeBuildInputs = [ cmake qttools ];
-
-  buildInputs = [
-    qtbase qtmultimedia qtx11extras qtwebengine
-    libidn qca-qt5 libXScrnSaver hunspell
-    libsecret libgcrypt libotr html-tidy libgpgerror libsignal-protocol-c
-    usrsctp
-  ] ++ lib.optionals voiceMessagesSupport [
-    gst_all_1.gst-plugins-base
-    gst_all_1.gst-plugins-good
-  ];
+    "-DCHAT_TYPE=${chatType}"
+    "-DENABLE_PLUGINS=${if enablePlugins then "ON" else "OFF"}"
+    "-DBUILD_PSIMEDIA=${if enablePsiMedia then "ON" else "OFF"}"
+  ] ++ (
+    assert builtins.all builtins.isString extraCmakeFlags; extraCmakeFlags
+  );
+
+  nativeBuildInputs =
+    [ cmake qttools ]
+    ++ lib.optional enablePsiMedia pkg-config;
```

Comment:
```suggestion
  nativeBuildInputs = [ cmake qttools ]
    ++ lib.optional enablePsiMedia pkg-config;
```

Reply:
@SuperSandro2000 if I compare this suggestion with another one:

``` nix
buildInputs = [
# ...
++ lib.optional (chatType == "webengine") qtwebengine;
```

``` nix
nativeBuildInputs = [ cmake qttools ]
  ++ lib.optional enablePsiMedia pkg-config;
```

They have different indentation level for their last lines. What is the intention of these suggestions?

Reply:
I don't know 🤔 I would just use nixpkgs-fmt and use that.

Reply:
@SuperSandro2000 if I apply `nixpkgs-fmt` on this file these line are staying unaffected. I hope that’s okay that I decline these suggestions. They only problem with them is that they are inconsistent with each other, I couldn’t put some logic behind them.

Reply:
nixpkgs-fmt does not enforce the new lines to prevent to many changes. I will not remember this conversation and the next time anything changes in this file I will probably come up with the formatting again. Just follow it, it is used everywhere in nixpkgs and does not cause unnecessary merge conflicts.

Reply:
@SuperSandro2000 sorry, I’m missing the point. Here is the diff that `nixpkgs-fmt` against that file gives me:
``` diff
diff --git a/pkgs/applications/networking/instant-messengers/psi-plus/default.nix b/pkgs/applications/networking/instant-messengers/psi-plus/default.nix
index a6a27c7fc0f..14f5a3a1c98 100644
--- a/pkgs/applications/networking/instant-messengers/psi-plus/default.nix
+++ b/pkgs/applications/networking/instant-messengers/psi-plus/default.nix
@@ -1,7 +1,21 @@
-{ lib, mkDerivation, fetchFromGitHub, cmake
-, qtbase, qtmultimedia, qtx11extras, qttools
-, libidn, qca-qt5, libXScrnSaver, hunspell
-, libsecret, libgcrypt, libotr, html-tidy, libgpgerror, libsignal-protocol-c
+{ lib
+, mkDerivation
+, fetchFromGitHub
+, cmake
+, qtbase
+, qtmultimedia
+, qtx11extras
+, qttools
+, libidn
+, qca-qt5
+, libXScrnSaver
+, hunspell
+, libsecret
+, libgcrypt
+, libotr
+, html-tidy
+, libgpgerror
+, libsignal-protocol-c
 , usrsctp
 
 , chatType ? "basic" # See the assertion below for available options
@@ -10,19 +24,19 @@
 
 , enablePlugins ? true
 
-# Voice messages
+  # Voice messages
 , voiceMessagesSupport ? true
 , gst_all_1
 
 , enablePsiMedia ? false
 , pkg-config
 
-, extraCmakeFlags ? [] # In addition to existing ones
+, extraCmakeFlags ? [ ] # In addition to existing ones
 }:
 
 assert builtins.elem (lib.toLower chatType) [
-  "basic"     # Basic implementation, no web stuff involved
-  "webkit"    # Legacy one, based on WebKit (see https://wiki.qt.io/Qt_WebKit)
+  "basic" # Basic implementation, no web stuff involved
+  "webkit" # Legacy one, based on WebKit (see https://wiki.qt.io/Qt_WebKit)
   "webengine" # QtWebEngine (see https://wiki.qt.io/QtWebEngine)
 ];
 
@@ -56,9 +70,19 @@ mkDerivation rec {
 
   buildInputs =
     [
-      qtbase qtmultimedia qtx11extras
-      libidn qca-qt5 libXScrnSaver hunspell
-      libsecret libgcrypt libotr html-tidy libgpgerror libsignal-protocol-c
+      qtbase
+      qtmultimedia
+      qtx11extras
+      libidn
+      qca-qt5
+      libXScrnSaver
+      hunspell
+      libsecret
+      libgcrypt
+      libotr
+      html-tidy
+      libgpgerror
+      libsignal-protocol-c
       usrsctp
     ]
     ++ lib.optionals voiceMessagesSupport [
```

Reply:
@SuperSandro2000 
1. `nixpkgs-fmt` does not change the style of indentation
2. `nixpkgs-fmt` touches other stuff as you can see
3. > Just follow it, it is used everywhere in nixpkgs and does not cause unnecessary merge conflicts.

   The style I used also targeted to avoid merge conflicts. To be patch friendly. I showed you some examples that are not obvious how to approach. For instance that one with the lack of indentation before `++`, there would be a merge conflicts when a change comes.

   ``` nix
   buildInputs = [
   # ...
   ++ lib.optional (chatType == "webengine") qtwebengine;
   ```

   Consider this if it was fixed and another entry was added:

   ``` nix
   buildInputs = [
     # ...
   ] ++ lib.optional (chatType == "webkit") qtwebkit
     ++ lib.optional (chatType == "webengine") qtwebengine
     ++ lib.optionals foo [
     # ...
   ];
   ```

   Imagine the last record appeared the latest. Isn’t this looks weird?

4. If you show me the logic, the standard, I will follow. What I see now is 2 suggestions that are in conflicts with each other as they are using different indentation style.

Reply:
@SuperSandro2000 
1. I changed that style to one that you proposed and I also replaced all `lib.optional` to `lib.optionals` even if it’s only one element inside, thus there are no more weird indentation cases, and also less potential merge conflicts in the future
2. I also applied `nixpkgs-fmt` on that file